### DISCUSSION TITLE
Is it possible to get JPEG XL images working?

### DISCUSSION THREAD

**Discussion by DevonAA** [stashapp/stash#3324] [General]
It seems stash app does not recognize the JPEG XL image format.

With "jxl" added as image extension the log shows:
`error processing "<path>.jxl": decoding image file "<path>.jxl": image: unknown format`

I've converted all my images from JPEG to JPEG XL a long time ago and saved 1 TB+ disc space with zero (lossless) quality loss.
Will definitely not convert back to an older worse format.

From what I understand the stash app uses FFMPEG. Good!
A quick internet search tells me FFMPEG added JPEG XL support ca. 6 month ago - so that should be no problem - I've updated it a few days ago.

Because I use FFMPEG for many things I just copy the exe it into C:\Windows.
Not sure if stash uses that one or maybe it comes with it somehow packed into "stash-win.exe" !?

Any idea how I can verify what FFMPEG version stash app is using? (To check for JPEG XL support)

Online JPEG XL converter for tests: [https://jpegxl.io/](https://jpegxl.io/)

**Comment by DogmaDragon** [comment]
You should have `ffmpeg-release-essentials.zip` in your `.stash` directory. Attaching the [README.txt](https://github.com/stashapp/stash/files/10327534/README.txt) with the details.

The format seems to be very new and has not even reached v1. Comparing it to WebP it saves about 7% more space but takes 8 times longer. 

If it's a matter of enabling a flag on ffmpeg I don't see why it couldn't be added, but I think the problem would be that none of the browsers support it natively without extra flags. <https://caniuse.com/jpegxl>

**Comment by DevonAA** [comment]
I found out stash was already using the ffmpeg.exe I placed in C:\Windows.

If I renamed that one and restarted stash it was detected as missing and started downloading the `ffmpeg-release-essentials.zip`.
But that zip-version is an older stable release not compiled with `--enable-libjxl`. So I reverted by to my latest ffmpeg.

I have verified my ffmpeg version can encode and decode JPEG XL images.
However stash still fails with the same error in the log. Apparently it's not a ffmpeg problem then.
Guess I have to dig in the code.

Btw. started testing the JPEG XL very early - December 2019.
Browsers and OS (thumbnails) has been configured to support JPEG XL for the last 2.5 years.
The bitstream was frozen 2 years ago. Converted a least 500k images since then.
From my point of view it's stable and safe to use despite it's version.

I use JPEG XL for the lossless JPEG recompression feature. Encode from JPG to JXL, save 20% in size.
If for whatever reason there is a problem - convert back to 100% same JPEG.
Perfect to archive with zero quality compromise and good compression.

**Reply by DogmaDragon** [reply]
Create a feature request. My only concern would again be browser support. Pale Moon is the only browser that supports it without needing to enable a flag. Which already is a very niche browser. 

![image](https://user-images.githubusercontent.com/103123951/210152650-cf79190e-b403-4f47-a2fc-9c11fa193fbc.png)

**Comment by feederbox826** [comment]
https://github.com/stashapp/stash/issues/4044
